tests/unit/aiplatform/test_cloud_training_job.py

```diff
@@ -0,0 +1,224 @@
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_RUNNING,
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_SUCCEEDED,
+            ),
+        ]
+        yield get_custom_job_mock
+
+
+def _get_custom_job_proto(state=None, name=None, error=None, version="v1"):
+    custom_job_proto = copy.deepcopy(_TEST_BASE_CUSTOM_JOB_PROTO)
+    custom_job_proto.name = name
+    custom_job_proto.state = state
+    custom_job_proto.error = error
+
+    if version == "v1beta1":
+        v1beta1_custom_job_proto = gca_custom_job_v1beta1.CustomJob()
+        v1beta1_custom_job_proto._pb.MergeFromString(
+            custom_job_proto._pb.SerializeToString()
+        )
+        custom_job_proto = v1beta1_custom_job_proto
+        custom_job_proto.job_spec.tensorboard = _TEST_TENSORBOARD_NAME
+
+    return custom_job_proto
+
+
+class LinearRegression(VertexModel): 
+ 
+        # constraint on no constructor arguments
+        def __init__(self):
+            input_size = 10
+            output_size = 10
+            super(LinearRegression, self).__init__()
+            self.linear = torch.nn.Linear(input_size, output_size)
+
+        def forward(self, x):
+            return self.linear(x)
+
+        def train_loop(self, dataloader, loss_fn, optimizer):
+            size = len(dataloader.size)
+            for batch, (X, y) in enumerate(dataloader):
+                pred = self.predict(X)
+                loss = loss_fn(pred, y)
+
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+
+        def fit(self, dataset):
+            loss_fn = nn.CrossEntropyLoss()
+            optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+            for t in range(epochs):
+                self.train_loop(data, loss_fn, optimizer)
+
+
+class TestCloudModelClass:
+
+    def test_create_cloud_class(self):
+        aiplatform.init(
+            project=_TEST_PROJECT,
+            location=_TEST_LOCATION,
+            staging_bucket=_TEST_STAGING_BUCKET,
+            encryption_spec_key_name=_TEST_DEFAULT_ENCRYPTION_KEY_NAME,
+        )
+
+        my_model = LinearRegression()
+        my_model.training_mode = 'cloud'
+
+        assert(my_model != None)
+
+    def test_fit_cloud_class(self, create_custom_job_mock, get_custom_job_mock):
+
+        aiplatform.init(
+            project=_TEST_PROJECT,
+            location=_TEST_LOCATION,
+            staging_bucket=_TEST_STAGING_BUCKET,
+            encryption_spec_key_name=_TEST_DEFAULT_ENCRYPTION_KEY_NAME,
+        )
+
+        my_model = LinearRegression()
+        my_model.training_mode = 'cloud'
+        my_model.fit(pd.DataFrame())
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
We should be testing that the API is called. For example:

```
create_custom_job_mock.assert_called_with(...)
```

This test will also likely not pass as the tmpdir created here is not necessarily the same as the one created by the implementation.

Reply:
Could you point me to a similar example of testing that an API is called? I'll look through other tests in the meantime as well.

Reply:
Could you possibly point me towards an example of a test that checks whether or not an API is called? I'll look through the other tests in the folder as well.

Reply:
Something like this should work: https://github.com/googleapis/python-aiplatform/blob/master/tests/unit/aiplatform/test_custom_job.py#L228